snooty/types.py

```diff
@@ -82,65 +105,141 @@ def from_docutils(cls, docutils_level: int) -> "Diagnostic.Level":
             level = max(level, cls.info)
             return cls(level)
 
-    severity: Level
-    message: str
-    start: Tuple[int, int]
-    end: Tuple[int, int]
+    @property
+    # why is the below incorrect?
+    # def severity(self) -> Level:
```

Comment:
It needs to be `-> "Diagnostic.Level":`, with quotes: this is a side-effect of bolting static types onto a language over 25 years after the language was born. The name `Diagnostic` hasn't actually been created yet, and class attributes cannot access other class attributes at time of instantiation